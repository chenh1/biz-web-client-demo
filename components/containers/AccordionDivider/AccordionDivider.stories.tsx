import type { AccordionDividerProps } from './AccordionDivider';
import React from 'react';
import { AccordionDivider } from './AccordionDivider';
import { Paragraph } from '../../core/Paragraph/Paragraph';
import { Heading } from '../../core/Heading/Heading';
import { Button } from '../../core/Button/Button';

type Args = {
  args: AccordionDividerProps
}

const Template = (args: AccordionDividerProps) => (
  <AccordionDivider className={"max-w-xl"} {...args}>
    <Paragraph>This is some text for a sample paragraph element</Paragraph>
    <Button label="CLICK ME" primary/>
  </AccordionDivider>
)

export default {
  component: Template
}

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const AccordionDividerExpandsWithContent: Args = { args: 
  { label: <Heading type="h2">There's some content below here</Heading>, dividerShouldExpand: true } 
};

export const AccordionDividerStaysAtLabel: Args = { args: 
  { label: <Heading type="h2">There's some content below here</Heading> } 
};

