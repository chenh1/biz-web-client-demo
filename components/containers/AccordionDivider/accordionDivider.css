.dividerShouldExpand {
  transform: translateY(-100%) translateY(1px);
}

.triangle.accordion { 
  width: 20px;
  transform-origin: center -250%;
  transition: .25s ease-in-out;
}

.triangle.up {
  transform: translateY(4px) rotate(0deg);
  display: block;
}

.triangle.down {
  transform: translateY(4px) rotate(180deg);
  display: block;
}

.triangle>div { 
  width: 20px;
  height: 2px;
  background: #050E1C;
}

.dark .triangle>div {
  background: #F3F4F6;
}

.triangle>div:before {
  content: " ";
  display: block;
  width: 20px;
  height: 2px;
  background: #050E1C;
  -webkit-transform: rotate(56deg);
  -moz-transform: rotate(56deg);
  -ms-transform: rotate(56deg);
  transform: rotate(56deg);
  position: absolute;
  top: -8px;
  right: -5px;
}
.dark .triangle>div:before {
  background: #F3F4F6;
}
.triangle>div:after {
  content: " ";
  display: block;
  width: 20px;
  height: 2px;
  background: #050E1C;
  -webkit-transform: rotate(-56deg);
  -moz-transform: rotate(-56deg);
  -ms-transform: rotate(-56deg);
  transform: rotate(-56deg);
  position: absolute;
  top: -8px;
  left: -5px;
}
.dark .triangle>div:after {
  background: #F3F4F6;
}