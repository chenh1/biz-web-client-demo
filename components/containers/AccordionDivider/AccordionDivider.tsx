'use client';

import React, { useState, ReactNode } from 'react';
import { Divider } from '../../core/Divider/Divider';
import { Box } from '../Box/Box';
import "./accordionDivider.css";

export type AccordionDividerProps = {
  className?: string;
  label: ReactNode;
  dividerShouldExpand?: boolean;
  children?: ReactNode;
  iconShow?: ReactNode;
  iconHide?: ReactNode;
  alwaysPushTriangle?: boolean;
}

export const AccordionDivider: React.FC<AccordionDividerProps> = ({
  className = "",
  label,
  dividerShouldExpand = false,
  children,
  iconShow,
  iconHide,
  alwaysPushTriangle
}) => {
  const [showContent, setShowContent] = useState<boolean>(false);

  return (
    <Box padding="0" className={`${className}`}>
      <div
        data-testid="accordionLabelClick"
        className={`cursor-pointer flex justify-between ${alwaysPushTriangle ? '' : 'md:justify-start'}`}
        onClick={() => setShowContent(!showContent)}
      >
        {label}
        {showContent ? iconHide : iconShow}
        {!iconHide && !iconShow &&
          <div className="relative w-5 ml-4">
            <div className={`triangle accordion ${showContent ? 'up' : 'down'}`}>
              <div></div>
            </div>
          </div>
        }
      </div>
      {!dividerShouldExpand && <Divider orientation="horizontal" />}
      <div
        className={`relative overflow-hidden duration-300 ease-in-out ${showContent ? 'h-full' : dividerShouldExpand ? 'h-[1px]' : 'h-0'}`}
      >
        <Box
          padding="0"
          gap="sm"
          className={`duration-300 ease-in-out ${showContent ? 'translate-y-0' : dividerShouldExpand ? 'dividerShouldExpand' : '-translate-y-full'}`}
        >
          {children}
          {dividerShouldExpand && <Divider orientation="horizontal" />}
        </Box>
      </div>
    </Box>
  );
};
