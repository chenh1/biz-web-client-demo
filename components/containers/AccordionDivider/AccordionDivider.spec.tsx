import { AccordionDivider } from "./AccordionDivider";
import { fireEvent, render } from '@testing-library/react';
import { JSDOM } from "jsdom"
const dom = new JSDOM()
global.document = dom.window.document
global.window = dom.window


it('renders the accordion divider', async () => {
  const { getByTestId, asFragment } = render(
    <AccordionDivider label={<h3>Label</h3>} />
  );

  expect(asFragment()).toMatchSnapshot();
});

it('renders opened accordion divider once label is clicked', async () => {
  const { getByTestId, asFragment } = render(
    <AccordionDivider label={<h3>Label</h3>} />
  );

  fireEvent.click(getByTestId("accordionLabelClick"))

  expect(asFragment()).toMatchSnapshot();
});