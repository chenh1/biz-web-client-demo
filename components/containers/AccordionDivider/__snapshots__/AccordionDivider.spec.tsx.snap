// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders opened accordion divider once label is clicked 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-4  p-2 "
  >
    <div
      class="cursor-pointer flex justify-between md:justify-start"
      data-testid="accordionLabelClick"
    >
      <h3>
        Label
      </h3>
      <div
        class="relative w-5 ml-4"
      >
        <div
          class="triangle accordion up"
        >
          <div />
        </div>
      </div>
    </div>
    <hr
      class="w-full h-0 border-0 border-t-[1px] border-night-black-300 dark:border-max-white "
    />
    <div
      class="relative overflow-hidden duration-300 ease-in-out h-full"
    >
      <div
        class=" grid  gap-4  p-2 duration-300 ease-in-out translate-y-0"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders the accordion divider 1`] = `
<DocumentFragment>
  <div
    class=" grid  gap-4  p-2 "
  >
    <div
      class="cursor-pointer flex justify-between md:justify-start"
      data-testid="accordionLabelClick"
    >
      <h3>
        Label
      </h3>
      <div
        class="relative w-5 ml-4"
      >
        <div
          class="triangle accordion down"
        >
          <div />
        </div>
      </div>
    </div>
    <hr
      class="w-full h-0 border-0 border-t-[1px] border-night-black-300 dark:border-max-white "
    />
    <div
      class="relative overflow-hidden duration-300 ease-in-out h-0"
    >
      <div
        class=" grid  gap-4  p-2 duration-300 ease-in-out -translate-y-full"
      />
    </div>
  </div>
</DocumentFragment>
`;
